script/playercomponent.js

```diff
@@ -126,10 +127,31 @@ define(
       function setCurrentTime (time) {
         userInteracted = true;
         if (transitions().canBeginSeek()) {
-          playbackStrategy.setCurrentTime(time);
+          if (windowType !== WindowTypes.STATIC && getLiveSupport(device) === LiveSupport.RESTARTABLE) {
+            reloadMediaElement(time);
+          } else {
+            playbackStrategy.setCurrentTime(time);
+          }
         }
       }
 
+      function reloadMediaElement (time) {
+        function doSeek (time) {
+          var thenPause = playbackStrategy.isPaused();
+          tearDownMediaElement();
+          if (time > (bigscreenPlayerData.time.windowEndTime - bigscreenPlayerData.time.windowStartTime) / 1000 - 10) {
```

Comment:
Assuming you will revisit this depending on whether my suggestion for seekable range works? You might want to use the `playbackStrategy.getSeekableRange()` to determine whether or not to set a start time. And you might not want to give it an arbitrary 10 second cushion.